main.go

```diff
@@ -145,10 +151,21 @@ func runFetch(c *cli.Context) error {
 		}
 	}
 
-	specific, desiredTag := isTagConstraintSpecificTag(options.TagConstraint)
+	var specific bool
+	var desiredTag string
+	var tagConstraint string
+
+	if options.GitRef != "" {
+		specific, desiredTag = isTagConstraintSpecificTag(options.GitRef)
+		tagConstraint = options.GitRef
+	} else {
+		specific, desiredTag = isTagConstraintSpecificTag(options.TagConstraint)
+		tagConstraint = options.TagConstraint
+	}
```

Comment:
do you always need to provide one of these two? - e.g. a ref or a tagConstraint

Reply:
Okay, had to re-familiarize myself with the code a bit to get you what I believe is the right answer here.

You can now specify --ref instead of --commit, --branch, or --tag. If you specify ref, then we detect whether it might be a tag constraint and capture that. If you specify a tag, then we detect that constraint and capture that. It's confusing, but I think it becomes less confusing if/when we deprecate --commit, --branch, and --tag and just have everyone use --ref.

I can't yet comment on the timeline on such a deprecation, since I don't know who is relying on `fetch` and what automation tools might need to change by doing such a deprecation.

Reply:
Resolving this, but if you'd like more discussion about it, reopen.